lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py

```diff
@@ -522,19 +523,22 @@ def subnets_to_associate(nacl, client, module):
     params = list(module.params.get('subnets'))
     if not params:
         return []
-    if params[0].startswith("subnet-"):
+    all_found = []
+    if any(x.startswith("subnet-") for x in params):
         try:
             subnets = client.describe_subnets(Filters=[
                 {'Name': 'subnet-id', 'Values': params}])
+            all_found.extend(subnets.get('Subnets', []))
         except botocore.exceptions.ClientError as e:
-            module.fail_json(msg=str(e))
-    else:
+            module.fail_json(msg=str(e), exception=traceback.format_exc())
```

Comment:
should we just convert this to use AnsibleAWSModule with module.fail_json_aws ?

Reply:
This PR is a smallish bugfix, so I didn't want to introduce the full switch. Converting to AnsibleAWSModule is on the list though :+1: